CONTRIBUTING.md

```diff
@@ -0,0 +1,119 @@
+
+- GHC 8.10.x (default)
+- GHC 9.2.x
+
+Upon entering a development environment for the first time, `nix` may ask you to activate the necessary option and the IOG nix cache (see `nixConfig` attribute in [`flake.nix`](./flake.nix)).
+
+#### Direnv
+
+[direnv](https://direnv.net/) is recommended, [install it for your OS and hook it to your shell](https://direnv.net/docs/installation.html).
+
+Then
+
+```console
+$ direnv allow
+```
+
+##### Use alternative compiler
+
+Check available [extra-compilers](./cells/prelude/haskell.nix#L6) (eg. `ghc924`) then:
+
+```console
+$ export DEVSHELL_TARGET=ghc924
+$ direnv reload
+```
+
+or permanently:
+
+```console
+$ echo DEVSHELL_TARGET=ghc924 >> .envrc.local
+$ direnv reload
+```
+
+#### Nix develop
+
+use `nix develop` or `nix develop .#ghc924` to enter a development shell.
+
+## Development workflows
+
+#### Available tools
+
+Available tools (`cabal`, `ghcid`, `hlint`, `weeder`, etc.) are displayed upon enterring the development shell. Type `menu` to see them again.
+
+Adding or modifying the list of development tools can be done from [devshellProfiles.nix](./cells/prelude/devshellProfiles.nix).
+
+#### Build and test
+
+using cabal:
+
+```console
+$ cabal update
+$ cabal build cardano-prelude:lib:cardano-prelude
+$ cabal run cardano-prelude-test:test:prelude-tests
+```
+
+or `std` (nix):
+
+```
+$ source <(std _carapace) # Optional, for completion.
+$ std //prelude/packages/cardano-prelude:lib:cardano-prelude:build
+$ std //prelude/apps/cardano-prelude-test:test:prelude-tests:run
+```
+
+(the std/nix build will use default compiler).
+
+#### Commits
+
+1.  Format the source tree with `treefmt` (which use [`ormolu`](https://ormolu-live.tweag.io/))
+2.  Use [conventionnal](https://www.conventionalcommits.org/) commit message (`<type>[(<scope>)]: <short description>]). Available commit types are from [std presets](https://std.divnix.com/reference/presets/nixago/conform.html).
+
+Otherwise [lefthook](https://github.com/evilmartians/lefthook) will prevent the commit to proceed (it will also format the code, so that you can retry to commit immediately).
+
+#### CI
+
+Running CI jobs locally can be done with [tullia](https://github.com/input-output-hk/tullia):
+
+```console
+$ tullia run ci
+```
+
+If this does not work (still WIP for local execution), then use `std` to build the required nix job:
+
+```console
+$ std //_automation/ciJobs/required:build
+```
+
+#### Updating cabal index-state
+
+If you need to update [cabal.project](./cabal.project) `index-state`, you need to first update the nix pin for hackage:
+
+```console
+$ nix flake lock --update-input haskellNix
+```
+
+Then run this to find-out last index available from this new pin:
+
+```console
+$ system=$(nix eval --raw --impure --expr builtins.currentSystem)
+$ nix eval --raw .#$system.prelude.library.last-index-state
+```
+
+and copy the output date as `index-state` value in `cabal.project`.
```

Comment:
I think this is backwards IMO: usually what I do is to update the `index-state` to whatever I want, and then just update `hackage.nix`. No need to ask hackage.nix for the latest index state or anything.